cores/arduino/stm32/analog.c

```diff
@@ -110,10 +112,12 @@
   * @}
   */
 
-/** @addtogroup STM32F4xx_System_Private_Variables
-  * @{
-  */
+ADC_HandleTypeDef AdcHandleTab[ADC_NUM];
+static DMA_HandleTypeDef DmaHandleTab[ADC_NUM] = {NULL};
+static PinName g_current_pinTab[ADC_NUM];
 static PinName g_current_pin = NC;
+void (*cb[ADC_NUM])(void *user_data);
+static bool use_dma = false;
```

Comment:
why isn't a "use_dma" for each ADC like use_dma[ADC_NUM] ?

Reply:
Here we don't need a table. I explain myself. When you want to use ADC peripheral with or without DMA, the same function was called : HAL_ADC_MspInit(). In this function I do the DMA part only if the call come from the dma function (adc_read_value_dma). The boolean dma_use fulfilled this role.

Reply:
so does it cover the case where ADC_1 is used with DMA and ADC_2 is used without DMA ?

Reply:
yes